### DISCUSSION TITLE
Countdown functionality (auto-off) on device which currently reports as Tuya TS011F (TS011F_plug_1 / _TZ3000_typdpbpg)

### DISCUSSION THREAD

**Discussion by JayFoxRox** [arendst/Tasmota#19993] [Q&A]
Hi,

I'm not using Tasmota (yet?) but I have a question in the same vein as https://github.com/arendst/Tasmota/discussions/14727 .

We have a heater which becomes dangerously hot after a couple of minutes.
As a potential solution, I started looking at sockets / smartplugs with countdown functions, so you can enable a device for "at most 5 minutes" (auto-off after X minutes).

To my surprise I came across various plugs which judging by store photos looked like the Aubess ZigBee smartplugs I already have (which is clearly a TuYa device) some of which claim to have a "countdown"-feature on their store page.. although I'm not sure whether this is what I'm looking for or whether my plugs expose this functionality.

We are already using our existing Aubess ZigBee smartplugs with zigbee2mqtt.
They expose as https://www.zigbee2mqtt.io/devices/TS011F_plug_1.html#tuya-ts011f_plug_1 (and look like the photo)
However, they don't seem to expose this "countdown" functionality (I'm not even sure if it exists in hardware).

I also found existing discussions about this in Tasmota (but not in stock firmware for newer devices which don't use the ESP8266/ESP32 anymore):

- https://github.com/arendst/Tasmota/discussions/13660
- https://github.com/arendst/Tasmota/discussions/13314
- https://github.com/arendst/Tasmota/discussions/10503
- ...

I also considered implementing this turn-off functionality in node-red, but it feels too dangerous (in case our zigbee coordinator crashes or hangs before it can turn off the heater plug).
I'd rather have a hardware based solution that's not dependent on some network (like Tasmota has).

So my questions:
- Did anybody reverse engineer it yet (does the Tuya firmware have it)?
    - How does Tuya implement this functionality (~ what's the API / protocol)?
    - How to control it through `zbsend`?
- Is there any info on firmware hacking for this particular device / leaked code, or would Tasmota run on it?
    - Could I reflash my device to another firmware which *has* this functionality? (even if not ESP8266 / ESP32 based?)

I could imagine this also exists in dimmers (turn off light after X minutes), so maybe Tuya has some common endpoint for this which works with this plug?

**Comment by JayFoxRox** [comment]
I have opened the device and it contained a TuYa ZT2S: https://developer.tuya.com/en/docs/iot/zt2s-module-datasheet?id=Kas9gdtath9p0

So I decided to just buy a handful of Tasmota plugs and use them with `pulsetime` now.

I'd still appreciate any more info about these TuYa plugs because I still have a bunch of them.

**Comment by Jason2866** [comment]
This is Tasmota discussions ;-)  We are replacing Tuya firmware, by replacing hardware.
For zigbee this is not possible. So only possible try to get it working as it is or better never buy Tuya devices.
In other words, you walk alone with Tuya zigbee...

**Reply by sfromis** [reply]
Of course, **if** the specific Zigbee end device already has such a feature, it could become a question of how to address that from Tasmota.
